open_bus_gtfs_etl/gtfs_loader.py

```diff
@@ -111,10 +132,15 @@ def same_route_exist_yesterday_adjust_dates(session: Session, route_from_gtfs):
     return True
 
 
+def _get_valid_stops_for_date(session: Session, date_to_analyze: date):
+    stops = session.query(Stop).filter(Stop.max_date <= date_to_analyze).order_by(Stop.max_date).all()
```

Comment:
you are only getting older dates then date_to_analyze, won't this be a problem if it runs for a past date?
please make sure the code supports running for past dates too, not only for latest date

Reply:
I think that we need to talk about this slowly changing dimension more - There are some cases we both don't handle well. I suggest to separate the duplicate within an execution and duplicate between executions. the bug here was about duplicates within one execution - when two lines use same stop the stop was created twice.

in response to your example, the code will fetch the stop item with max date 20.10 and in case all fields are the same, no new record will be created - just the max date will be modified. 

i looked in the following SIRI etl logic and there you are fetching only items that belongs to the specific date. correct me if i'm wrong but there if we will use your use case - since no stop will be fetched - a new record will be created. 
```  
elif objname == 'stop':
            recorded_at_time = kwargs['recorded_at_time']
            return session.query(Stop).filter(
                Stop.min_date <= recorded_at_time.date(),
                recorded_at_time.date() <= Stop.max_date,
                Stop.is_from_gtfs == False,
                Stop.code == int(kwargs['stop_point_ref'])
            ).one_or_none()
```

Reply:
@AvivSela this bug is fixed?

Reply:
@OriHoch, yes, instead of using max date, I'm using min date , so stops that their min date was yesterday and max date will be tomorrow are included